ports/raspberrypi/boards/pimoroni_inky_frame_5_7/board.c

```diff
@@ -100,10 +111,10 @@ void board_init(void) {
         false,  // color_bits_inverted
         0x000000,  // highlight_color
         refresh_sequence, sizeof(refresh_sequence),
-        28.0,  // refresh_time
+        2.0,  // refresh_time
```

Comment:
I don't think this is correct. It needs to be long enough for the sequence to complete before we send it more commands. Usually the busy pin would minimize this but it isn't used here.

I timed it at just under 28 seconds.

Reply:
Currently, I need to do this:

  1. call display.refresh()
  2.    _clean_area()
  3.    wait until refresh_time expires
  4.    update areas
  5.    return immediatly
  6. call time.sleep(until refresh is really finished)
  7. set board.ENABLE_DIO = 0 (shutdown and cut power)

During the wait-time after _clean_area (3) the display does nothing, it just wastes battery with a tight loop. (6) is necessary regardless of the value of refresh_time, since there is no wait after update areas.

So although the refresh time is really about 30 seconds, in the current implementation it makes more sense to set it to something short and wait after display.refresh() returns (I currently wait seconds_per_frame). 

The better solution would be that display.refresh() returns after refresh is finished and power can be cut without problems. But until then, a long refresh time has negative impact.

Reply:
I don't think anything happens during (3) even when the refresh_time is 28 secs. The display flashes two times after sending the clean-area bytes (dimming all colors), but that is it. The real refresh flashes much more often and the display is still busy way after the refresh() method returns.

And I don't get the point of waiting half a minute after _clean_area, but not waiting that time after the real screen update. This does not make sense. And in all my testing I haven't seen any problems with the two seconds refresh_time.

To repeat the point: I do believe the necessary refresh  time is about 30 seconds. But this is the time after the refresh() method returns and the display is still busy refreshing. But the refresh_time parameter has nothing to do with the refresh behavior.